### ISSUE TITLE
TFLite conversion fails when empty weights are involved

### ISSUE THREAD

**Issue by leandro-gracia-gil** [tensorflow/tensorflow#53113]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- TensorFlow installation (pip package or built from source): pip
- TensorFlow library (version, if pip package or github SHA, if built from source): TF 2.7 and tf-nightly are both affected

### 2. Code
```python
import tensorflow as tf                                                        
                                                                               
class TestModel(tf.keras.models.Model):                                        
                                                                               
  def __init__(self, dims, **kwargs):                                          
    super().__init__(**kwargs)                                                 
    self._dense = tf.keras.layers.Dense(dims)                                  
        
  @tf.function
  def test(self, x):
    return self._dense(x)                                                      
                                                                               
  
# Fails only if dims=0.
test_model = TestModel(dims=0)                                                 
signatures = [
  test_model.test.get_concrete_function(x=tf.TensorSpec([None, 10], tf.float32))  
]                                                                              
    
converter = tf.lite.TFLiteConverter.from_concrete_functions(signatures, test_model)                                                                            
converter.optimizations = [tf.lite.Optimize.DEFAULT]                           
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS]         

# Fails with "ConverterError: Quantize weights transformation failed."         
# But calling directly test_model.test(tf.random.normal(shape=[1, 10]) works.
tflite_model = converter.convert()
  
interpreter = tf.lite.Interpreter(model_content=tflite_model)                  
result = interpreter.get_signature_runner()(x=tf.random.normal(shape=[1, 10]))
```

### 3. Failure after conversion
Conversion fails when empty tensors are involved. Doing the same manually with an empty matrix variable has the same effect.
`tensorflow.lite.python.convert_phase.ConverterError: Quantize weights transformation failed.`

This does not happen when calling the model method directly, which correctly returns an empty tensor.
It does not fail either if the dense layer weights are not empty by setting `dims > 0`.

For context, I'm not trying to figure out corner cases that fail. I have a multiple signature model where some training time settings can cause some weights to become empty. This is not a problem with regular TF, but this limitation of TFLite is forcing me to introduce additional code in multiple places to work around the issue.

**Comment by sushreebarsa** [comment]
@jvishnuvardhan Was able to replicate the issue on colab using TF v2.7.0 and tf-nightly,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/a8042f5a279da16599c2860b1f5a0c3e/53113.ipynb) for reference.Thanks!

**Comment by leandro-gracia-gil** [comment]
I can confirm the problem is fixed in the latest tf-nightly build (2.10.0-dev20220420).
I'll be closing this then. Thanks!
